airflow/decorators/__init__.py

```diff
@@ -29,9 +30,11 @@ def __getattr__(self, name):
         if name.startswith("__"):
             raise AttributeError(f'{type(self).__name__} has no attribute {name!r}')
         decorators = ProvidersManager().taskflow_decorators
-        if name not in decorators:
-            raise AttributeError(f"task decorator {name!r} not found")
-        return decorators[name]
+        if name in decorators:
+            return decorators[name]
+        if name == "sensor":
+            return sensor
```

Comment:
@dimberman can you comment on whether this is a good approach or whether instead we should do multiple inheritance with mixins as is done with `@task.virtualenv`?  i lean towards supporting the approach taken here but interested if you want to advocate for the mixin approach

Reply:
Can provider implement `sensor` decorator? If yes in that case that one will be used. I'm wondering if we should not initialise `ProviderManager` with `_taskflow_decorators` that would include predefined builtin decorators (or add builtins in the same way)?  In this way we may probably use only one mechanism of registering decorators and skip mixin approach.

Reply:
Hi @turbaszek, I agree that it is ideal to have a single mechanism to register sensor decorators, but the one defined in this PR is for the base sensor operator which is not in the provider package.

Reply:
@mingshi-wang if my provider package will define `sensor` decorator it will override the builtin one - is this expected behaviour?  I wouldn't say so as this imposes some security risk in my opinion (I think I'm using airflow sensor but under the hood something else is being executed).

Reply:
@turbaszek You are correct and this is a valid concern to me. A solution I can think of is to raise an error in the ProviderManager if "sensor" is registered by any provider. This makes "sensor" a reserved name but I would like to hear you suggestion.

Reply:
I think the same goes for other builtins (python for example). We either register them in `ProviderManager` upfront and raise an error when adding provider decorators or check here in init for possible name clash.

@dimberman @kaxil @potiuk happy to hear your opinion here 🤔